docs/dev/plugins.md

```diff
@@ -328,6 +328,62 @@ no_command do
 end
 ```
 
+## Implementing Reporter Plugins
```

Comment:
This new informative documentation is in a dev doc. Does this need to be moved to something more public-facing?

Reply:
Yes, it will land in the Plugins section I would guess

Reply:
It is documentation for developers, and is linked from the public-facing Plugins page. I wasn't planning on moving it, this is just adding a section to an existing doc.

Reply:
@clintoncwolfe Oh! I had forgotten/assumed dev docs were mostly internal similar to other chef repos. (That's good to know as we migrate the docs.) Thank you for clarifying. With that context, this new content should stay where it is. :) Copyedits will appear soon.